src/test_l1_tvSettings.c

```diff
@@ -2079,91 +2113,103 @@ void test_l1_tvSettings_negative_SaveBacklight (void)
+	result = SaveBacklight(videoSource,pqValue,(tvVideoFormatType_t)-1,50);
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
 
 	/* Step 09: Calling tvsettings SaveBacklight and expecting the API to return tvERROR_INVALID_PARAM */
-	result = SaveBacklight((tvVideoSrcType_t)Configfile.videoSrcStruct.videoSourceValue[0],Configfile.picmodeStruct.pqValue[0],(tvVideoFormatType_t)Configfile.videoFormtStruct.videoFormatValue[0],-1);
+	result = SaveBacklight(videoSource,pqValue,videoFormat,-1);
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
 
 
 	/* Step 10: Calling tvsettings SaveBacklight and expecting the API to return tvERROR_INVALID_PARAM */
-	result = SaveBacklight((tvVideoSrcType_t)Configfile.videoSrcStruct.videoSourceValue[0],Configfile.picmodeStruct.pqValue[0],(tvVideoFormatType_t)Configfile.videoFormtStruct.videoFormatValue[0],101);
+	result = SaveBacklight(videoSource,pqValue,videoFormat,101);
 	UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
 
 	/* Step 11: Calling tvsettings SaveBacklight and expecting the API to return tvERROR_INVALID_PARAM */
+	count = UT_KVP_PROFILE_GET_LIST_COUNT("tvSettings/VideoSource/index");
 	for(int i =VIDEO_SOURCE_ALL ; i < VIDEO_SOURCE_MAX; i++)
 	{
 		SupportAvailable = false;
-		for(int j =0 ; j < Configfile.videoSrcStruct.size; j++)
+		for(int j =0 ; j < count; j++)
 		{
-			if(Configfile.videoSrcStruct.videoSourceValue[j] == i)
+			snprintf(keyValue, UT_KVP_MAX_ELEMENT_SIZE, "tvSettings/VideoSource/index/%d", j);
+			videoSource = (tvVideoSrcType_t)UT_KVP_PROFILE_GET_UINT32(keyValue);
+			if(videoSource == i)
 			{
 				SupportAvailable = true;
 				break;
 			}
 		}
 
 		if(!SupportAvailable){
-			result = SaveBacklight((tvVideoSrcType_t)i,Configfile.picmodeStruct.pqValue[0],(tvVideoFormatType_t)Configfile.videoFormtStruct.videoFormatValue[0],50);
+			result = SaveBacklight(videoSource,pqValue,videoFormat,50);
 			UT_ASSERT_AUTO_TERM_NUMERICAL(result, tvERROR_INVALID_PARAM);
 		}
 	}
 
 	/* Step 12: Calling tvsettings SaveBacklight and expecting the API to return tvERROR_INVALID_PARAM */
+	count = UT_KVP_PROFILE_GET_LIST_COUNT("tvSettings/PictureMode/index");
+
+	videoSource = (tvVideoSrcType_t)UT_KVP_PROFILE_GET_UINT32("tvSettings/VideoSource/index/0");
 	for(int i =0 ; i < PQ_MODE_MAX; i++)
 	{
 		SupportAvailable = false;
-		for(int j =0 ; j < Configfile.picmodeStruct.size; j++)
+		for(int j =0 ; j < count; j++)
 		{
-			if(Configfile.picmodeStruct.pqValue[j] == i)
+			snprintf(keyValue, UT_KVP_MAX_ELEMENT_SIZE, "tvSettings/PictureMode/index/%d", j);
+			pqValue = (int)UT_KVP_PROFILE_GET_UINT32(keyValue);
+			if(pqValue == i)
 			{
 				SupportAvailable = true;
 				break;
 			}
 		}
 
 		if(!SupportAvailable){
-			result = SaveBacklight((tvVideoSrcType_t)Configfile.videoSrcStruct.videoSourceValue[0],i,(tvVideoFormatType_t)Configfile.videoFormtStruct.videoFormatValue[0],50);
+			result = SaveBacklight(videoSource,i,videoFormat,50);
```

Comment:
can you add a /* TODO: Review this and similar tests, all functionality must be called to the backend API, it's the API that should decide if it's supported or not, not the test */

Reply:
Added